sdk/python/feast/infra/online_stores/snowflake.py

```diff
@@ -0,0 +1,234 @@
+
+            dfs[i] = df
+
+        if dfs:
+            agg_df = pd.concat(dfs)
+
+            # This combines both the data upload plus the overwrite in the same transaction
+            with get_snowflake_conn(config.online_store, autocommit=False) as conn:
+                write_pandas_binary(
+                    conn, agg_df, f"[online-transient] {config.project}_{table.name}"
+                )  # special function for writing binary to snowflake
+
+                query = f"""
+                    INSERT OVERWRITE INTO "{config.online_store.database}"."{config.online_store.schema_}"."[online-transient] {config.project}_{table.name}"
+                        SELECT
+                            "entity_feature_key",
+                            "entity_key",
+                            "feature_name",
+                            "value",
+                            "event_ts",
+                            "created_ts"
+                        FROM
+                          (SELECT
+                              *,
+                              ROW_NUMBER() OVER(PARTITION BY "entity_key","feature_name" ORDER BY "event_ts" DESC, "created_ts" DESC) AS "_feast_row"
+                          FROM
+                              "{config.online_store.database}"."{config.online_store.schema_}"."[online-transient] {config.project}_{table.name}")
+                        WHERE
+                            "_feast_row" = 1;
+                """
+
+                conn.cursor().execute(query)
+
+            if progress:
+                progress(len(data))
+
+        return None
+
+    @log_exceptions_and_usage(online_store="snowflake")
+    def online_read(
+        self,
+        config: RepoConfig,
+        table: FeatureView,
+        entity_keys: List[EntityKeyProto],
+        requested_features: List[str],
+    ) -> List[Tuple[Optional[datetime], Optional[Dict[str, ValueProto]]]]:
+        assert isinstance(config.online_store, SnowflakeOnlineStoreConfig)
+
+        result: List[Tuple[Optional[datetime], Optional[Dict[str, ValueProto]]]] = []
+
+        with get_snowflake_conn(config.online_store) as conn:
+
+            df = (
+                conn.cursor()
+                .execute(
+                    f"""
+                SELECT
+                    "entity_key", "feature_name", "value", "event_ts"
+                FROM
+                    "{config.online_store.database}"."{config.online_store.schema_}"."[online-transient] {config.project}_{table.name}"
+                WHERE
+                    "entity_feature_key" IN ({','.join([('TO_BINARY('+hexlify(serialize_entity_key(combo[0])+bytes(combo[1], encoding='utf-8')).__str__()[1:]+")") for combo in itertools.product(entity_keys,requested_features)])})
+            """,
+                )
+                .fetch_pandas_all()
+            )
+
+        for entity_key in entity_keys:
+            entity_key_bin = serialize_entity_key(entity_key)
```

Comment:
```suggestion
            entity_key_bin = serialize_entity_key(entity_key, entity_key_serialization_version=2)
```

Reply:
Ideally you should be using the version from the repo_config but since this is new I think `2` may be better.